src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// See https://aka.ms/ms-id-web/identity-federation.
+    /// </summary>
+    public class AzureFederatedTokenProvider
+    {
+        /// <summary>
+        /// See https://aka.ms/ms-id-web/identity-federation.
+        /// </summary>
+        /// <param name="federatedClientId"></param>
+        public AzureFederatedTokenProvider(string? federatedClientId)
+        {
+            _federatedClientId = federatedClientId;
+            ClientAssertionProvider = GetSignedAssertionFromFederatedTokenProvider;
+        }
+
+        private readonly string? _federatedClientId;
+
+        /// <summary>
+        /// Prototype of certificate-less authentication using a signed assertion
```

Comment:
```suggestion
        /// Certificate-less authentication using a signed assertion
```

Reply:
I tried to commit your suggestion directly but it wouldn't take. oh well. thanks again.